src/main/scala/Day1/Iteration4.scala

```diff
@@ -0,0 +1,50 @@
+package Day1
+
+import scala.annotation.tailrec
+
+object Iteration4 extends App {
+
+  def repeater(name: String, input: Int) = {
+
+    @tailrec
+    def repeaterRec(name: String, end: Int, count: Int = 1): Unit = {
+      println(name)
+      if(count<end)
+        repeaterRec(name, end, count+1)
+    }
+
+    repeaterRec(name, input.abs)
+  }
+
+  def squareRepeater(name: String, count: Int) ={
+
+    @tailrec
+    def squareRepeaterRec(name: String, end: Int, count: Int = 1):Unit = {
+      println(name.repeat(end))
+      if(count<end)
+        squareRepeaterRec(name, end, count+1)
+    }
+
+    squareRepeaterRec(name, count)
+  }
+
+  def fizzBuzz(word1: String, word2: String, count: Int): Unit ={
+    @tailrec
+    def fizzBuzzRec(word1: String, word2: String, end: Int, count: Int = 1): Unit ={
```

Comment:
could make this more optimal by saving the result values for `count%3==0` and `count%5==0`